tests/functional/styleguide.js

```diff
@@ -0,0 +1,81 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public" + '/n' +
+ "* License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/* global casper */
+'use strict';
+
+var config = require('../lib/config');
+var helpers = require('../lib/helpers');
+var path = '/styleguide/';
+var url = config.base() + path;
+
+casper.test.begin('Styleguide, Elements: ' + url, 3, function suite(test) {
+    casper.start(url, function() {
+        test.assertHttpStatus(200);
+
+        test.assertResourceExists(/media\/img\/styleguide\/home/);
+        test.assertVisible('#sidebar', 'The sidebar navigation is visible');
+    });
+
+    casper.run(function() {
+        test.done();
+        helpers.done();
+    });
+});
+
+casper.test.begin('Styleguide, Navigation flow: ' + url, 11, function suite(test) {
```

Comment:
Feels like there's a lot going on in this one test? Might it be better to split this up a bit?

I don't think we gain much by clicking a link, waiting for another page to load and then checking the URL. It would be much faster just to check that the link's href is what we expect? If we want to test something at that URL, then make it it's own test? Doing this for a few key style guide pages would be enough I think.